internal/transactions/storage.go

```diff
@@ -0,0 +1,248 @@
+package transactions
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+const (
+	// Redis key prefixes
+	prefixTx        = "tx:"     // For storing transaction data
+	prefixIndex     = "idx:"    // For storing indexes
+	prefixGroup     = "group:"  // For storing grouped transactions
+	prefixStats     = "stats:"  // For storing statistics
+	prefixFilter    = "filter:" // For storing filter results
+	prefixMetadata  = "meta:"   // For storing transaction metadata
+	prefixAddresses = "addr:"   // For storing address-related data
+)
+
+// Storage handles all Redis operations for transactions
+type Storage struct {
+	rdb *redis.Client
+}
+
+// NewStorage creates a new storage instance
+func NewStorage(rdb *redis.Client) *Storage {
+	return &Storage{rdb: rdb}
+}
+
+// StoreTransaction stores a transaction with its metadata
+func (s *Storage) StoreTransaction(ctx context.Context, tx *StoredTransaction) error {
+	// Store the full transaction data
+	txKey := fmt.Sprintf("%s%s:%d", prefixTx, tx.Metadata.From, tx.Metadata.Nonce)
+	txData, err := json.Marshal(tx)
+	if err != nil {
+		return fmt.Errorf("error marshaling transaction: %w", err)
+	}
+
+	// Store metadata separately for efficient filtering
+	metaKey := fmt.Sprintf("%s%s:%d", prefixMetadata, tx.Metadata.From, tx.Metadata.Nonce)
+	metaData, err := json.Marshal(tx.Metadata)
+	if err != nil {
+		return fmt.Errorf("error marshaling metadata: %w", err)
+	}
+
+	// Store in Redis
+	pipe := s.rdb.Pipeline()
+	pipe.Set(ctx, txKey, txData, 24*time.Hour)
```

Comment:
I was thinking of having two different queues. "pending" and "queued".

That way our Redis Queue would look like

```
{
  pending: {
    txKey: txData,
  }
  queued: {
    txKey: txData
  }
}
```

We can achieve that using HSET. The listName is already passed on to the `processTransactionBatch` method.